jina/executors/__init__.py

```diff
@@ -335,31 +335,29 @@ def save(self, filename: str = None) -> bool:
         :param filename: file path of the serialized file, if not given then :attr:`save_abspath` is used
         :return: successfully persisted or not
         """
-        if not self.is_updated:
-            self.logger.info(f'no update since {self._last_snapshot_ts:%Y-%m-%d %H:%M:%S%z}, will not save. '
+        if not self.read_only and self.is_updated:
+            self.is_updated = False
```

Comment:
Maybe move this to the end of this `if`, so that if an error happens and the data isn't saved, the `is_updated` stays `True`